### DISCUSSION TITLE
DXMT July Status: Ready-to-Use For Certain Games

### DISCUSSION THREAD

**Discussion by 3Shain** [3Shain/dxmt#9] [Announcements]
`DXMT` is a Metal-based translation layer for Direct3D 11. It currently allows running D3D11 applications on macOS with the aid of `Wine`.

After another one month of development, we are proudly to introduce the new progress we've achieved.

## Games Primarily Get Supported
- Final Fantasy XIV: Dawntrail
![](https://github.com/user-attachments/assets/cc92137f-aa5c-498d-9d7f-61920b7cfecf)
> screenshoot from CHCOOBOO#5205
- Genshin Impact
![](https://github.com/user-attachments/assets/4b1362f3-89ad-4752-93cd-9f46358bc5aa)
> screenshoot from Skadi#6969
- Zenless Zone Zero
![](https://github.com/user-attachments/assets/341e3274-3b72-4f81-93a4-47e95600b1fc)

> screenshoot from Skadi#6969

- A space fantasy RPG that I can't write down the name since running it on macOS doesn't make any sense due to the anti-cheat component but it works anyway
![](https://github.com/user-attachments/assets/70613abe-3417-4b5d-8496-176711aec8ce)

> screenshoot from tonyy#7168

The very reason these games get focused is __there exists dedicated projects__ ([XIV on Mac](https://www.xivmac.com/), [Yet Another Anime Games Launcher(Yaagl)](https://github.com/yaagl/yet-another-anime-game-launcher)) that help MacOS users play these games. So users don't have to worry about building from source and patching Wine, they just click and run.  This helps us easily get feedbacks and diagnose errors from end-users. Besides, they are all popular games that has a good amount of players who have a variety of Mac devices, which gives us more idea about the capability as well as compatibility of DXMT.

DXMT is already presented on [XIV on Mac](https://www.xivmac.com/) for several days before this announcement posted. And users generally report they've noticed __a significant performance boost__ comparing to the old DXVK+MoltenVK solution. Not only it performs great (smooth as butter) on high-end Mac models, but also makes the game much more playable on models with only 8GB RAMs. Besides performance, the rendering is also stable and flawless (after day and night debugging on page fault). The devs of XIV on Mac initially wanted to find an alternative of their original graphics backend to solve visual glitches occurred on recent expansion, and now the problem get perfectly solved, nice!

And there is an interesting experimental feature that worth highlighting, the _MetalFX Spatial Upscaling_, that can generate more pixels with a small performance overhead. __It is opt-in game-agnostic, everything runs on DXMT can be benefited__. 

![](https://github.com/user-attachments/assets/2ba081be-2453-4ad1-9bfd-bcb8baf3afdb)

By adjusting system UI scale (sometime it's switch called High Resolution Mode/Retina Mode/macOS Scaling) and DXMT MetalFX scale factor, you can either have the game output pixel-perfect resolution for lower hardware cost, or get an effect of supersampling anti-aliasing.

If you happen to be a FFXIV player or you are interested in the game, you can download XIV on Mac from [their website](https://www.xivmac.com/) and join their discord server.

---
For all other games that are planned to be _DXMT-powered_ in future [Yaagl](https://github.com/yaagl/yet-another-anime-game-launcher) updates, the internal testing also reflects good results.  For many scenes that DXVK+MoltenVK or D3DMetal strive to get high and stable frame-rate, DXMT just yields consistent 60fps. One technical thing worth mentioning is that they are (customized different versions of) Unity Engine games and uses GPU-Skinning feature that requires Stream-Output (or Transform Feedback for OpenGL/Vulkan) which has no counterpart in Metal, but we are still able to hack for a specific case when rasterization is disabled and primitive type is non-strip and more technical assumptions that usually hold. As a result, they just work out-of-the-box. In the future we will switch to mesh shader to emulate geometry pipeline, providing a complete implementation for good.

&nbsp; | &nbsp;
:-------------------------:|:-------------------------:
![](https://github.com/user-attachments/assets/00eace35-3def-4a49-8759-1154d79afa50) |  ![](https://github.com/user-attachments/assets/f146d80b-efcc-40db-b701-3694a4180aa1)
> screenshoot from Anmeifan & Suze#3188

Join [Yaagl Discord Server](https://discord.gg/HrV52MgSC2) for their up-to-date information.

## Roadmap

Of course we plan to support more games (actually we didn't test other games, so some games might have been "supported" but we don't know). Specifically, these features will be focused and actively worked on:

- Tessellation pipeline
- Non-Apple GPUs support
- Persistent shader caches (yes it's performing great even without a cache!)
- DeferredContext
- Better DXGI implementation (for fullscreen, vsync, hdr, etc.)

And instructions of testing DXMT with other games will be released later, however there is no ETA at the moment.

## ... and one more thing
@3Shain is collaborating with CodeWeavers from August, 2024. They will support the development of DXMT, just like how they've been supporting Wine! And if everything goes well, DXMT would become a new option of CrossOver, to provide a better experience including but not limited to gaming on mac.

**Comment by AnotiaWang** [comment]
Awesome work!

**Comment by Ghussy** [comment]
Amazing! Very cool 😎

**Comment by osekom** [comment]
Since I came across this repository and saw what was done I knew it wouldn't be long before this happened, I'm 90% sure that another product like croosover and whyski are interested

**Comment by josepheiba** [comment]
yo cool, I will test it for Master Duel

**Comment by AndrewTriesToCode** [comment]
To add another data point I tried it with xcom2 wotc and the title screen loads fine. Right away I can tell the performance is better than DXVK+MoltenVK or D3DMetal. 

However, starting a game crashes as the scene loads up.

**Reply by 3Shain** [reply]
If you want me to take a look at the issue, please create a #reports thread on the Discord server.
